src/odoc/compilation_unit.ml

```diff
@@ -14,34 +14,54 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
+open Odoc_model
 open Or_error
 
-type t = Odoc_model.Lang.Compilation_unit.t
+type t =
+  | Page_content of Lang.Page.t
+  | Module_content of Lang.Compilation_unit.t
 
-let save file t =
+(** Written at the top of the files. Checked when loading. *)
+let magic = "odoc-%%VERSION%%"
+
+(** Exceptions while saving are allowed to leak. *)
+let save_unit file t =
   Fs.Directory.mkdir_p (Fs.File.dirname file);
   let oc = open_out_bin (Fs.File.to_string file) in
-  Root.save oc t.Odoc_model.Lang.Compilation_unit.root;
+  output_string oc magic;
   Marshal.to_channel oc t [];
   close_out oc
 
-let units_cache = Hashtbl.create 23 (* because. *)
+let save_page file page =
```

Comment:
It looks wrong that `Compilation_unit` is saving/loading pages.

Reply:
This module is now saving and loading `.odoc` files. How should we call it ?
I used the name "compilation_unit" for `Lang.Page | Lang.Compilation_unit` in the `Resolver` module, the second case is named `Module`. This will need to be changed too.

Reply:
`Documentation_unit` (and `t = Page | Module`) ?

Reply:
I've changed to unambiguous names. This module is now `Odoc_file` and this record is `Page_content ... | Unit_content ...`.

Reply:
"Unit" alone doesn't really mean much (except maybe the type). Either you put the whole thing ("Compilation unit") or choose another word.

Reply:
We commonly use the word "unit" for "compilation unit" already. `Compilation_unit_content of Lang.Compilation_unit.t` sounds unnecessary to me.